lib/inferno/dsl/fhir_evaluator/ig.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'rubygems/package'
+require 'zlib'
+
+module FhirEvaluator
+  # IG is a wrapper class around the relevant concepts inside an IG.
+  # Not everything within an IG is relevant to the evaluation this tool does.
+  class IG
+    attr_accessor :profiles, :extensions, :value_sets, :search_params, :examples
+
+    def initialize(ig_path)
+      raise "#{ig_path} is not an IG file" unless File.file?(ig_path)
```

Comment:
This constructor is doing way too much. It shouldn't be doing much more than assigning instance variables. Make a class method that initializes the object and then performs the operations rather than having it all done in the constructor.

Reply:
The file is deleted. IG import will be handled at phase 2.